### ISSUE TITLE
win32ui crashes on exit with large wx Application

### ISSUE THREAD

**Issue by ghost** [mhammond/pywin32#675]
Under Windows 8.1 / Python27 64Bit / pywin32 build 219 there seems to be a race condition during the shutdown of the application.

Debugging showed, that some Windows messages are sent AFTER the Python interpreter has already shut down (Py_Finalize already called).

The call to

```~~
//win32win.cpp:77 (inside Python_check_message)
CEnterLeavePython _celp;
```~~

is therefore working with a dead python instance and crashes.

Suggested fix:
The error can be prevented, if the following code is added to the win32uimodule.cpp file:

```~~
static void UnhookWindowsMessages() 
{
	if (hhook)
		UnhookWindowsHookEx(hhook);
	hhook = 0;
}

static void win32ui_finalize (void)
{
  UnhookWindowsMessages();
}

// At the end of PYWIN_MODULE_INIT_FUNC(win32ui):

  Py_AtExit(win32ui_finalize);
```~~

Working with Py_AtExit is not always the best choice, but in this case it seems legit.

Reported by: anode85

Original Ticket: [pywin32/bugs/675](https://sourceforge.net/p/pywin32/bugs/675)

**Comment by flutefreak7** [comment]
This seems to still be a problem.  The following is causing 2 Segfaults for me on application exit:

    from PyQt5 import QtWidgets
    import win32ui
    
    app = QtWidgets.QApplication([])
    window = QtWidgets.QLabel("Hello")
    window.show()
    app.exec_()

This is a minimal example for PyQt so it doesn't appear to matter how big the application is in this case.  If I comment out win32ui, it works fine.  I'm on Windows 10 64-bit, Python 3.6.

Also if I run this in PyCharm's debug mode, there is no segfault, so the PyCharm debugger is meddling with things enough to cause the race condition to not occur.

**Comment by mhammond** [comment]
Sadly, mixing UI toolkits in one app is always going to cause problems. I'd take a patch for this, but I'm not going to dig into it.

**Comment by mhammond** [comment]
Probably fixed by #1841
